fw/ParticleSystem.js

```diff
@@ -0,0 +1,203 @@
+    let toAdd;
+    if(rate === undefined) {
+        toAdd = amount;
+    } else {
+        toAdd = rate;
+    }
+    if (style === 'image') {
+        var iImage = graphics.Img(image);
+    }
+    that.render = function () {
+        for (let particle = 0; particle < particles.length; particle++) {
+            if (particles[particle].alive >= 100) {
+                if (particles[particle].style === 'circle') {
+                    graphics.drawCircle({
+                        x: particles[particle].particleCenter.x,
+                        y: particles[particle].particleCenter.y,
+                        radius: particles[particle].size,
+                        fill: particles[particle].fill,
+                        stroke: particles[particle].stroke
+                    });
+                } else if (style == 'image') {
+                    graphics.drawImage({
+                        image: iImage,
+                        dx: particles[particle].particleCenter.x,
+                        dy: particles[particle].particleCenter.y,
+                        dWidth: imagedWidth,
+                        dHeight: imagedHeight
+                    })
+                }
+            }
+        }
+    };
+
+    that.update = function (elapsedTime) {
+        let keepMe = [];
+
+        for (let particle = 0; particle < particles.length; particle++) {
+            particles[particle].alive += elapsedTime;
+            particles[particle].particleCenter.x += (elapsedTime * particles[particle].speed * particles[particle].direction.x);
+            particles[particle].particleCenter.y += (elapsedTime * particles[particle].speed * particles[particle].direction.y);
+            particles[particle].rotation += particles[particle].speed / .5;
+
+            if (particles[particle].alive <= particles[particle].lifetime) {
+                keepMe.push(particles[particle]);
+            }
+        }
+
+        for (let particle = 0; particle < toAdd; particle++) {
+            let p = {
+                particleCenter: {
+                    x: center.x,
+                    y: center.y
+                },
+                direction: nextCircleVector(angleOffset, angleTotal),
+                speed: Math.abs(nextGaussian(speedmean, speedstdev)), // pixels per millisecond
+                rotation: 0,
+                lifetime: nextGaussian(lifetimemean, lifetimestdev), // milliseconds
+                alive: 0,
+                size: nextGaussian(sizemean, sizestdev),
+                fill: fill,
+                stroke: stroke,
+                style: style
+            };
+            keepMe.push(p);
+            if(rate === undefined) {
+                toAdd--;
+            }
+        }
+        particles = keepMe;
+        if(particles.length === 0) {
+            return false;
+        }
+    };
+
+    return that;
+}
+
+function ParticleSystemManager() {
+    let that = {};
+    that.addParticleSystem = function (center, {
```

Comment:
The indentation here is a little confusing. There should be one more level of indentation for the addParticleSystem parameter list and function body.